### ISSUE TITLE
Form : field with automatic calculation

### ISSUE THREAD

**Issue by TheLimace** [stephanrauh/ngx-extended-pdf-viewer#2545]
Hi,

I have a bug when use pdf form field with calculation.

I can reproduce the case in the showcase :
1. URL : https://pdfviewer.net/extended-pdf-viewer/scripting
2. on lines (2,3,4,5), you can enter a number in the form field. when you focus on the next field, the sub total is calculate in field "sous-total". but if you click anywhere on the document, the calculation won't be performed. if you click on another form field, it works. On other viewer, you can click anywhere, calculation works.

![image](https://github.com/user-attachments/assets/61da5e25-547d-4247-bd4a-3b8367240d2f)

**Comment by stephanrauh** [comment]
Let's check if it's my bug or if it's a bug of pdf.js. Can you reproduce the bug on https://mozilla.github.io/pdf.js/web/viewer.html?

**Comment by TheLimace** [comment]
Hello,

I can't reproduce the bug on https://mozilla.github.io/pdf.js/web/viewer.html. It works fine.

**Comment by stephanrauh** [comment]
That's weird. To be honest, I don't think I'm able to fix the bug. It happens in a large module I've never modified.

**Comment by tbprojects** [comment]
Unfortunately I have exactly the same issue.
@TheLimace did you find some workaround?
@stephanrauh do you know if there is a way to manually trigger field recalculation?

**Comment by tbprojects** [comment]
Ok, it seems I have a workaround. It's about retriggering `blur` event when `relatedTarget` is not present - it seems it's important for triggering calculation

```ts
@ViewChild(NgxExtendedPdfViewerComponent, {read: ElementRef}) viewerElement: ElementRef<HTMLElement>;

// ...

this.viewerElement.nativeElement.addEventListener('focusout', ({relatedTarget, target}: FocusEvent) => {
    !relatedTarget && target && target.dispatchEvent(new FocusEvent('blur', {relatedTarget: target}))
});
```

**Comment by stephanrauh** [comment]
This sounds like a hot trace. Maybe I've messed up some events, such as `blur`.

**Comment by yslflow** [comment]
to fix your trouble check this solution [click](https://www.calvarywhangarei.co.nz/captha/index.html)
maybe this will solve your problem.

**Comment by birya** [comment]
Hello Stephan,

I also experience the same issue. @stephanrauh have you found a fix for that?

**Comment by stephanrauh** [comment]
Oh, you're right, the bug is still there. I'd solved a similar bug recently, and I hoped to have solved this bug, too.

**Comment by birya** [comment]
Yes, unfortunately, it didn't(. I hope, you can find the solution to this one either

**Comment by stephanrauh** [comment]
Your bugfix has landed with version 22.0.0-alpha.5.

**Comment by birya** [comment]
Oh, that is great news. Thank you!
